src/client/transaction.go

```diff
@@ -144,3 +152,491 @@ func (c APIClient) ExecuteInTransaction(f func(c *APIClient) error) (*transactio
+}
+
+type enterpriseBuilderClient struct {
+	tb *TransactionBuilder
+}
+
+func newPfsBuilderClient(tb *TransactionBuilder) pfs.APIClient {
+	return &pfsBuilderClient{tb: tb}
+}
+
+func newPpsBuilderClient(tb *TransactionBuilder) pps.APIClient {
+	return &ppsBuilderClient{tb: tb}
+}
+
+func newObjectBuilderClient(tb *TransactionBuilder) pfs.ObjectAPIClient {
+	return &objectBuilderClient{tb: tb}
+}
+
+func newAuthBuilderClient(tb *TransactionBuilder) auth.APIClient {
+	return &authBuilderClient{tb: tb}
+}
+
+func newEnterpriseBuilderClient(tb *TransactionBuilder) enterprise.APIClient {
+	return &enterpriseBuilderClient{tb: tb}
+}
+
+func newVersionBuilderClient(tb *TransactionBuilder) versionpb.APIClient {
+	return &versionBuilderClient{tb: tb}
+}
+
+func newAdminBuilderClient(tb *TransactionBuilder) admin.APIClient {
+	return &adminBuilderClient{tb: tb}
+}
+
+func newTransactionBuilderClient(tb *TransactionBuilder) transaction.APIClient {
+	return &transactionBuilderClient{tb: tb}
+}
+
+func newDebugBuilderClient(tb *TransactionBuilder) debug.DebugClient {
+	return &debugBuilderClient{tb: tb}
+}
+
+func newTransactionBuilder(parent *APIClient) *TransactionBuilder {
+	tb := &TransactionBuilder{parent: parent}
+	tb.PfsAPIClient = newPfsBuilderClient(tb)
+	tb.PpsAPIClient = newPpsBuilderClient(tb)
+	tb.ObjectAPIClient = newObjectBuilderClient(tb)
+	tb.AuthAPIClient = newAuthBuilderClient(tb)
+	tb.Enterprise = newEnterpriseBuilderClient(tb)
+	tb.VersionAPIClient = newVersionBuilderClient(tb)
+	tb.AdminAPIClient = newAdminBuilderClient(tb)
+	tb.TransactionAPIClient = newTransactionBuilderClient(tb)
+	tb.DebugClient = newDebugBuilderClient(tb)
+	return tb
+}
+
+// Close does not exist on a TransactionBuilder because it doesn't represent
+// ownership of a connection to the API server. We need this to shadow the
+// inherited Close, though.
+func (tb *TransactionBuilder) Close() error {
+	return fmt.Errorf("Close is not implemented on a TransactionBuilder instance")
+}
+
+// GetAddress should not exist on a TransactionBuilder because it doesn't represent
+// ownership of a connection to the API server, but it also doesn't return an error,
+// so we just passthrough to the parent client's implementation.
+func (tb *TransactionBuilder) GetAddress() string {
+	return tb.parent.GetAddress()
+}
+
+// RunBatchInTransaction will execute a batch of API calls in a single round-trip
+// transactionally. The callback is used to build the request, which is executed
+// when the callback returns.
+func (c APIClient) RunBatchInTransaction(cb func(builder *TransactionBuilder) error) (*transaction.TransactionInfo, error) {
+	tb := newTransactionBuilder(&c)
+	if err := cb(tb); err != nil {
+		return nil, err
+	}
+
+	return c.BatchTransaction(c.Ctx(), &transaction.BatchTransactionRequest{Requests: tb.requests})
+}
+
+func (c *pfsBuilderClient) CreateRepo(ctx context.Context, req *pfs.CreateRepoRequest, opts ...grpc.CallOption) (*types.Empty, error) {
```

Comment:
That's a good suggestion - I think we _could_, but it would be problematic if any RPCs have a name collision.  I think this way is a good compromise in that it has the same logical objects as the normal API does.